firmware/libraries/versavis/src/VN100.cpp

```diff
@@ -0,0 +1,191 @@
+////////////////////////////////////////////////////////////////////////////////
+//  VN100.cpp
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright (c) 2018 Flight Dynamics and Control Lab
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// NOTE: Before running this code, you must configure the IMU as
+// defined in the README file.
+////////////////////////////////////////////////////////////////////////////////
+
+#include "VN100.h"
+#include "helper.h"
+#include "versavis_configuration.h"
+
+VN100::VN100(ros::NodeHandle *nh, const String &topic, const int rate_hz,
+             Timer &timer)
+    : Imu(nh, topic, rate_hz, timer), kImuSyncTimeoutUs(4000) {
+  imu_accelerator_sensitivity_ = 1.0 / (0.00025 * 9.81);
+  imu_gyro_sensitivity_ = 1.0 / (0.05 * M_PI / 180);
+}
+
+void VN100::setup() {
+  DEBUG_PRINTLN((topic_ + " (VN100.cpp): Setup.").c_str());
+  // Start Serial1 for IMU communication
+  Serial1.begin(921600, SERIAL_8N1);
+  delay(20);
+  Serial.setTimeout(2);
+
+  // Check whether IMU is setup correctly. Configuration is done on VectorNav
+  // Control Center software.
+  Serial1.print("$VNRRG,00*XX\r\n");
+  delay(20);
+  byte response[100];
+  Serial1.readBytes(response, Serial1.available());
+
+  // Check that "User Tag" is set to 7665727361766973 (HEX for versavis).
+  if (strcmp((char *)response, "$VNRRG,00,7665727361766973*51\r\n") != 0) {
+    error((topic_ + " (VN100.cpp): IMU is not setup correctly.").c_str(), 11);
+  }
+  Imu::setupPublisher();
+}
+
+////////////////////////////////////////////////////////////////////////////
+// Read the IMU bytes.
+////////////////////////////////////////////////////////////////////////////
+int16_t *VN100::readImuData() {
+  clearBuffer();
+  Serial1.print("$VNBOM,1*XX\r\n");
+  const uint64_t local_tic = micros();
+  while (Serial1.available() < 30) {
```

Comment:
Why `30`? Maybe make this a verbose variable, i.e., `const int kImuMessageLength = 30`. Why constrain to maximum buffer size anyways? Shouldn't it be `31` then?

Reply:
Thanks for the suggestion. Its 30 though :-)